controller/api/models.py

```diff
@@ -298,6 +298,24 @@ def _start_containers(self, to_add):
         if set([c.state for c in to_add]) != set(['up']):
             err = 'warning, some containers failed to start'
             log_event(self, err, logging.WARNING)
+        # if the user specified a health check, try checking to see if it's running
+        try:
```

Comment:
Are we sure about that? I think one of the issues enumerated in #3813 is that the Docker proxy accepts connections before the container is even listening. We should test this to make sure that the first connect will block until we actually get a response, either good or bad.